src/borg/helpers.py

```diff
@@ -628,73 +676,93 @@ def _match(self, path):
         return (self.regex.search(path) is not None)
 
 
-_PATTERN_STYLES = set([
+_PATTERN_CLASSES = set([
     FnmatchPattern,
     PathFullPattern,
     PathPrefixPattern,
     RegexPattern,
     ShellPattern,
 ])
 
-_PATTERN_STYLE_BY_PREFIX = dict((i.PREFIX, i) for i in _PATTERN_STYLES)
+_PATTERN_CLASS_BY_PREFIX = dict((i.PREFIX, i) for i in _PATTERN_CLASSES)
 
-InclExclPattern = namedtuple('InclExclPattern', 'pattern ptype')
-RootPath = object()
-PatternStyle = object()
+CmdTuple = namedtuple('CmdTuple', 'val cmd')
 
 
-def get_pattern_style(prefix):
+class IECommand(Enum):
```

Comment:
Yes, I agree that IE is a little bit cryptic -- it could simply be InclExcl instead of IE, but I thought that made referencing the values kind of a lot to type.  I don't like "PatternType" because these really are more general commands (in some cases, of what should be done with a particular pattern).  Also, it can lead to confusion about what is meant by "pattern type" -- elsewhere in the code this would mean, for example, whether it is a `re` or `sh` pattern, etc.